memif-attach/main.go

```diff
@@ -223,6 +277,45 @@ func sendInitMsg(ctx context.Context, connClient io.Writer, helloMsg *HelloMsg)
 	log.Entry(ctx).Infof("Init message sent")
 }
 
+func createAndSendMemRegion(ctx context.Context, unixConn net.Conn, memifDetails memif.MemifDetails) *memfd.Memfd {
+	// crate share memory region
+	regionSize := getRegionSize(memifDetails)
+	mfd, err := memfd.Create()
+	if err != nil {
+		log.Entry(ctx).Fatalln("Memory region create Error", err)
+	}
+	setSizeErr := mfd.SetSize(int64(regionSize))
+	if setSizeErr != nil {
+		log.Entry(ctx).Fatalln("Memory region set size Error", err)
+	}
+	fd := mfd.Fd()
+
+	// init add region msg and send to master
+	Ack := ackMsg{0}
+	addRegionMsg := &AddRegionMsg{
+		Ack:   Ack,
+		Index: uint16(fd), // should fix later
+		Size:  regionSize, // calculate in https://github.com/FDio/vpp/blob/b8e7a45d56be9f3e11b07b82fd899160e2af1bf1/src/plugins/memif/memif.c#L376
+	}
+	buf := new(bytes.Buffer)
+	writeErr := binary.Write(buf, binary.BigEndian, addRegionMsg)
+	if writeErr != nil {
+		log.Entry(ctx).Fatalln("Encoding Error", err)
+	}
+
+	rights := syscall.UnixRights(int(fd))
+	unixConn.(interface {
+		WriteMsgUnix(b, oob []byte, addr *net.UnixAddr) (n, oobn int, err error)
+	}).WriteMsgUnix(buf.Bytes(), rights, nil)
```

Comment:
Now that my program can compile without error and run smoothly in sending the add region message, I wonder how I can check if region fd is being sent correctly to VPP Master?